src/rgw/rgw_lua_utils.cc

```diff
@@ -24,6 +27,61 @@ int RGWDebugLog(lua_State* L)
   auto message = luaL_checkstring(L, 1);
   ldout(cct, 20) << "Lua INFO: " << message << dendl;
   return 0;
+
+}
+
+int RGWDebugLogF(lua_State* L)
+{
+  auto cct = reinterpret_cast<CephContext*>(lua_touserdata(L, lua_upvalueindex(FIRST_UPVAL)));
+
+  fmt::dynamic_format_arg_store<fmt::format_context> store;
+  std::string format = luaL_checkstring(L, 1);
+
+  int num_args = 0;
+  size_t pstart = 0, pend = 0;
+
+  // Does not handle escaping or positional arguments
+
+  pend = format.find("{}");
+  while(pend != std::string::npos) {
+    num_args++;
+    pstart = pend + 1;
+    pend = format.find("{}", pstart);
+  }
+
+  auto num_vals = lua_gettop(L);
+  if (num_vals <= num_args) {
+    ldout(cct, 20) << "Lua ERROR: RGWDebugLogF:" << "Not enough arguments. num_args="
+                   << num_args << ", num_vals=" << num_vals << dendl;
+    return -1;
+  }
+
+  for (auto i = 1; i <= num_args; i++) {
+    switch (lua_type(L, i+1)) {
+      case LUA_TNUMBER:
+        store.push_back(lua_tonumber(L, i+1));
```

Comment:
i thought that `format_arg` is `std::variant` ?
why not pass in the number as-is without converting to a tring?

Reply:
Done.